stan/math/rev/arr/functor/coupled_ode_system.hpp

```diff
@@ -72,7 +72,7 @@ struct coupled_ode_system<F, double, var> {
       : f_(f),
         y0_dbl_(y0),
         theta_(theta),
-        theta_dbl_(value_of(theta)),
+        theta_copy_(theta),
```

Comment:
I discussed with @wds15. This is not the right copy; we need to create a copy that is disconnected from `theta` and is on our `no_chain` stack.

Reply:
Please test that there's nothing added to the stack on construction.

Reply:
Ok, I am using the nonstack thing now (really ugly to get things there!). However, I am not sure how to test and why to test that nothing gets on construction onto the stack. Ok, I could use `ChainableStack::instance().var_stack.size()`... but that is a not an exported API and is used nowhere in any test which I could find such that I am not sure that this can be tested given our API. If you think it is important to test, can you suggest a scheme to do so?

Reply:
Never mind... I figured that we can test the stack size changes if we nest things.

... this is one of the annoyances of our nesting system - things are slightly inconsistent wrt to the non-nested stack; at least to me.

So the coupled_ode_system is now tested to not put stuff on the AD stack upon construction.